src/gui/TrackContentObjectView.cpp

```diff
@@ -1456,8 +1417,54 @@ TimePos TrackContentObjectView::quantizeMarkerPos( TimePos midiPos, bool shiftMo
 	}
 	else
 	{
-		return (TimePos(midiPos + m_initialTCOPos).quantize( snapSize ) - m_initialTCOPos);
+		return TimePos(midiPos + m_initialTCOPos).quantize( snapSize ) - m_initialTCOPos;
+	}
+}
+
+
+
+
+// This should eventually be available for all clip types, hence why it's not
+// located inside SampleTCOView
+bool TrackContentObjectView::splitTCO( QMouseEvent * me )
```

Comment:
What about making `TrackContentObjectView::splitTCO` a virtual method on `TrackContentObjectView` and override it on `SampleTCOView`? Then if the split action is added to other types of TCOs we can just override them on the respective ones (i.e.: `PatternView`).

I also think it could be useful in the future if we had the relative X position as the parameter of the method call instead of the mouse event. Then we could call `splitTCO` in contexts other than a mouse event (just examples: a keyboard shortcut for splitting, or a built in splicer). The split call would then be `splitTCO(me->pos().x())`.

Reply:
Virtual is a good idea! I also agree that passing a mouse event feels excessive, but as it stands the method also uses other event properties, like modifers.

Reply:
True, I missed that the `unquantizationModHeld` and `quantizeMarkerPos` were also using it. You could still move away from using the event directly by having something like `splitTCO(const int xPos, const bool unquantize, const bool quantizeClosest)` then calling it as `splitTCO(me->pos().x(), unquantizeModHeld(me), me->modifiers() & Qt::ShiftModifier)`.